bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/PredefinedTypes.java

```diff
@@ -242,5 +246,8 @@
         TYPE_DETAIL = new BMapType(TypeConstants.MAP_TNAME, TYPE_CLONEABLE, EMPTY_MODULE);
         TYPE_ERROR_DETAIL = ReadOnlyUtils.setImmutableTypeAndGetEffectiveType(TYPE_DETAIL);
         TYPE_ERROR = new BErrorType(TypeConstants.ERROR, EMPTY_MODULE, TYPE_DETAIL);
+        ANY_AND_READONLY_OR_ERROR_TYPE =
+        ReadOnlyUtils.setImmutableTypeAndGetEffectiveType(new BUnionType(Arrays.asList(ANY_AND_READONLY_TYPE,
+                TYPE_ERROR)));
```

Comment:
We don't have to call `setImmutableTypeAndGetEffectiveType` here right? We just need the union. The union initialization logic will actually mark this as an immutable union, so IINM this call is unnecessary anyway.

```suggestion
        new BUnionType(Arrays.asList(ANY_AND_READONLY_TYPE, TYPE_ERROR));
```

Let's also fix the indentation here.